server/controllers/search-controller.js

```diff
@@ -2,21 +2,27 @@
 
 var router = require("express").Router();
 
-router.get("/", getWords);
+router.post("/", search);
 
-function getWords(req, res) {
-    var WordList = require("../models/word-list").WordList;
-    
-    // use distinct to return wordlist array
-    WordList.distinct("wordlist").lean().exec()
-        .then(function(results){
-            res.status(200).json(results);
-        })
-        .catch(function(err) {
-            console.log("Error: " + err);
-            res.status(500).send(err.message);
-        });
+function search(req, res) {
+    if(req.body.payload) {
+        var WordList = require("../models/word-list").WordList;
+        var regex = new RegExp("(\\b\\w*" + req.body.payload + "\\w*\\b)", "gi");
```

Comment:
I had to change the regex here for some reason. I don't know why. I struggled with it for hours. But here, it works differently than it did in the client. Again, no idea why. May look at it again later.